### PR TITLE
fix: resource definition batch resource operations fail for duplicate names

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

Resource definition detail page manage resources across different environments, which results in duplicate resource names failing on name checks.

**Solution:**

Change name checking to ID checking.

**Related Issue:**
#2032

### CODE REVIEW

**Comment by alexcodelf** [comment]
Changing names to IDs can result in failures when batch creating resources since we use names to establish dependencies. Can we conduct the name check on a project-by-project basis? In case there are multiple resources spread across different projects, can we check each one individually?

**Comment by gitlawr** [comment]
What happen when we clone multiple resources with dependencies? Previously, we get dependencies from attributes before relationships are stored in db.

File: pkg/apis/resource/common.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func DeleteResources(req CollectionDeleteRequest, mc model.ClientSet, kubeConfig
 
 		resources, err := tx.Resources().Query().
```
Comment (gitlawr) [inline pkg/apis/resource/common.go:32]: What about we group these resources by their environment and do TopologicalSort per group?

File: pkg/apis/resource/common.go
Comment (alexcodelf) [inline pkg/apis/resource/common.go:1]: Please also check collection upgrade.

File: pkg/apis/resource/common.go [hunk 2]
```diff
@@ -55,44 +49,83 @@ func DeleteResources(req CollectionDeleteRequest, mc model.ClientSet, kubeConfig
...
+		environmentIDToResources[r.EnvironmentID] = append(environmentIDToResources[r.EnvironmentID], r)
+	}
+
+	return environmentIDToResources, nil
+}
 
-		if err = upgrade(req.Context, kubeConfig, mc, entity, req.Draft); err != nil {
+func UpgradeResources(req CollectionRouteUpgradeRequest, mc model.ClientSet, kubeConfig *rest.Config) error {
+	err := mc.WithTx(req.Context, func(tx *model.Tx) error {
+		environmentIDToResources, err := getResourcesGroupByEnvironment(tx, req.IDs(), req.Context)
```
Comment (alexcodelf) [inline pkg/apis/resource/common.go:104]: The resources will not receive request resources update data `req.Model()` if we query from DB.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**